src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
+    def path: Path
+    def allBytes: Array[Byte]
+    def nodeType: fs.NodeType
+    def size: Long
+    def permissions: PosixFilePermissions
+    def atime: Long
+    def mtime: Long
+
+    def name : String    = path.getFileName.to_s
+    def blockCount: Long = (size + blockSize - 1) / blockSize
+    def blockSize: Long  = 512 // FIXME
+  }
+
+  // this could have been an apply method in an object allowing the use of Metadata instead of Metadataish
+  // I however could not get myself to repeat all those arguments
```

Comment:
Hmmm, but I am materializing data. I had `getData: () => Array[Byte]` in there. There should be a way to get the data from a `Metadataish` thingy. To keep things simple I just added `allBytes` because that was shared with `Pathish`.

I can rename `Metadata` to `NonPhysicalFile` though, but it needs some way to get the data. And to keep things simple (while conceptually incorrect) I just kept `allBytes`.

If you have better ideas, let me know.